cishouseholds/validate_class.py

```diff
@@ -92,14 +138,28 @@ def validate_column(self, operations):
                 else:
                     self.execute_check(check["function"], check["error_message"], column_name, list(method.values())[0])
 
-    def validate(self, operations):
+    def validate(self, operations: dict):
+        """
+        Executes validation by given logic validation to all columns. TODO
+        Parameters
+        ----------
+        operations
+        """
         for method, params in operations.items():
             if type(params) != list:
                 params = [params]
             for p in params:
                 self.execute_check(self.functions[method]["function"], self.functions[method]["error_message"], **p)
 
-    def validate_udl(self, logic, error_message, columns):
+    def validate_udl(self, logic: Any, error_message: str, columns: List[str]):
+        """
```

Comment:
logic: the pyspark logic to be applied (must return a true/false outcome to denote a pass or fail of check)
error_message: string pattern to be added to list of errors if the check fails
columns: list of columns used in the function